+pnpm.io

getneil opened the pull request:
[pnpm](https://pnpm.io/) a better and faster node package manager

Commit:
add pnpm of pnpm.io

Commit:
move pnpm to its own folder

Commit:
copy artifacts to dist

Commit:
test dir

mxcl commented:
Like our packaging guide recommends, see what brew do: https://github.com/Homebrew/homebrew-core/blob/master/Formula/pnpm.rb

mxcl commented on projects/pnpm.io/package.yml:
```diff
@@ -0,0 +1,26 @@
+distributable:
+  url: https://github.com/pnpm/pnpm/archive/refs/tags/v{{ version }}.tar.gz
```
use: `https://registry.npmjs.org/pnpm/-/pnpm-{{version}}.tgz` per brew formula

mxcl commented on projects/pnpm.io/package.yml:
```diff
@@ -0,0 +1,26 @@
+    curl.se: '>=5'
+    nodejs.org: '>=16'
+  script:
+    curl -L https://get.pnpm.io/v6.16.js | node - add --global pnpm@next-7 npm@7
```
Seems like the tarball is in a ready to execute format: https://github.com/Homebrew/homebrew-core/blob/master/Formula/pnpm.rb#L29-L33

getneil replied:
thank you for this big help.

mxcl commented on projects/pnpm.io/package.yml:
```diff
@@ -0,0 +1,26 @@
+
+build:
+  dependencies:
+    curl.se: '>=5'
```
don't forget to drop this

Commit:
copy brew install strategy

Commit:
add install prefix

Commit:
try different install process

Commit:
retry add prefix

Commit:
set PNPM_HOME var

Commit:
add path

Commit:
test run export path only

Commit:
rm mkdir

getneil committed:
provide cjs files pnpm.io

getneil committed:
install node_modules

getneil committed:
add npmjs.com dep

mxcl commented on projects/pnpm.io/package.yml:
```diff
@@ -0,0 +1,26 @@
+    npm install
+    export PNPM_HOME=root/Library/pnpm
+    export PATH="$PNPM_HOME:$PATH";
+    export PATH="$(pwd)/bin:$PATH";
```
your script doesn't move anything to `{{prefix}}` so nothing is installed. Confused what you're doing anyway, all brew does is move a few files: https://github.com/Homebrew/homebrew-core/blob/master/Formula/pnpm.rb#L29-L33

getneil replied:
sorry did not fully understand the purpose of {{prefix}}, i just wrongly assumed pwd would be where to just put stuff. but thanks im finally getting a different error now.

getneil committed:
install on prefix path

getneil committed:
test cp install

Commit:
test uncertainly strip version

Commit:
provide fixed version

Commit:
fixed version in url

Commit:
undo versions fixed

Commit:
no proper pnpx test

getneil commented on projects/pnpm.io/package.yml:
```diff
@@ -0,0 +1,26 @@
+distributable:
+  # url: https://registry.npmjs.org/pnpm/-/pnpm-{{version}}.tgz
+  url: https://registry.npmjs.org/pnpm/-/pnpm-7.18.2.tgz
```
i think pnpm is returning a bunch of weird versions with chars.

Commit:
pnpx is actually deprecated: use undefined
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Cannot read property 'includes' of undefined or

Commit:
cleanup setup and dependency

Commit:
test if working correctly

Commit:
rm unused line

mxcl commented:
Excellent!

mxcl commented:
I do wonder if you were building locally first? That way is much faster than using CI/CD to evaluate build steps.

getneil commented:
> I do wonder if you were building locally first? That way is much faster than using CI/CD to evaluate build steps.

i tried first trying to run it directly failed on that, tried to dockerize also (encountered setting up linux on m1)
then i tried to run the entire pipeline using https://github.com/nektos/act the only part that run was the `get-diff`
then I tried using my mac mini intel there some parts started working correctly for the linux part

mxcl commented:
what did you try? What failed?

If you have issues with tea please tell us about them here or in Slack, if you are having issues then hundred other people also are and they also aren't reporting them.
